app/build.gradle

```diff
@@ -17,6 +17,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    useLibrary 'org.apache.http.legacy'
```

Comment:
regarding permission, according to google's storage documentation we don't need the permission if we only store private files, which are not going to be used by other applications, in case of offline data it's indeed not needed, but we are also going to have the pdf files, which have to be opened by other applications, so we actually do need the permission to store the files in public directories. unless there's a way to let other applications open our private files...